src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -119,19 +124,21 @@ public override string[] GetEnumNames()
         public override Array GetEnumValues()
         {
             if (!IsActualEnum)
-                throw new ArgumentException(SR.Arg_MustBeEnum, "enumType");
-
-            // Get all of the values
-            ulong[] values = Enum.InternalGetValues(this);
-
-            // Create a generic Array
-            Array ret = Array.CreateInstance(this, values.Length);
+                ThrowMustBeEnum();
 
+            // Get all of the values as the underlying type and copy them to a new array of the enum type.
+            Array values = Enum.GetValuesAsUnderlyingTypeNoCopy(this);
+#if MONO
+            // TODO https://github.com/dotnet/runtime/issues/79224:
```

Comment:
The ifdef is around `Array.CreateInstance`. It should be around `Array.Copy` instead.

Reply:
Yup. My local build agrees with you; apparently I didn't push my ammended commit.